### PR TITLE
ENG-6224 Create a separate distribution to run Deploy worker

### PR DESCRIPTION
**Pull request by vpugar-digital**

### CODE REVIEW

File: src/main/groovy/ai/digital/integration/server/tasks/worker/SetWorkersLogbackLevelsTask.groovy [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import ai.digital.integration.server.util.LogbackConfigs
+import ai.digital.integration.server.util.WorkerUtil
+import groovy.xml.QName
+import groovy.xml.XmlUtil
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+
+import static ai.digital.integration.server.constant.PluginConstant.PLUGIN_GROUP
+
+class SetWorkersLogbackLevelsTask extends DefaultTask {
```
Comment (acierto) [inline src/main/groovy/ai/digital/integration/server/tasks/worker/SetWorkersLogbackLevelsTask.groovy:15]: Code is almost identical.
Let's reuse the code

File: src/main/groovy/ai/digital/integration/server/IntegrationServerPlugin.groovy [hunk 2]
```diff
@@ -84,8 +88,12 @@ class IntegrationServerPlugin implements Plugin<Project> {
         project.tasks.create(StartSatelliteTask.NAME, StartSatelliteTask)
 
         //Workers
+        project.tasks.create(SyncServerPluginsWithWorkerTask.NAME, SyncServerPluginsWithWorkerTask)
```
Comment (acierto) [inline src/main/groovy/ai/digital/integration/server/IntegrationServerPlugin.groovy:91]: New tasks has to be documented in `tasks.md`

File: src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        WorkerUtil.getWorkers(project).forEach { Worker worker ->
+            copyServerDirToWorkerDir(worker)
+        }
+    }
+
+    void copyServerDirToWorkerDir(Worker worker) {
+        if (!WorkerUtil.isExternalRuntimeWorker(worker, project)) {
+            def sourceDir = ServerUtil.getServerWorkingDir(project)
+            def destinationDir = Paths.get(WorkerUtil.getWorkerWorkingDir(worker, project)).toFile()
+            destinationDir.setExecutable(true)
```
Comment (acierto) [inline src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy:40]: It's not clear for me why full folder has to be executable.
I think it has to be only files from `bin` folder

File: src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            ProcessUtil.chMod(project, "755", "${destinationDir.getAbsolutePath().toString()}")
+
+            project.logger.lifecycle("Copy Worker ${worker.name} runtime from ${sourceDir} to ${destinationDir}")
```
Comment (acierto) [inline src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy:43]: We copy only plugins, we have to highlight this.

File: src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            // delete plugins from zip
+            FileUtils.deleteDirectory(Paths.get(WorkerUtil.getWorkerWorkingDir(worker, project), "plugins").toFile())
+
+            [
+                    "hotfix",
+                    "plugins"
+            ].forEach { String dir ->
+                FileUtils.copyDirectory(
+                        Paths.get(ServerUtil.getServerWorkingDir(project), dir).toFile(),
+                        Paths.get(WorkerUtil.getWorkerWorkingDir(worker, project), dir).toFile()
```
Comment (acierto) [inline src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy:54]: Usually by convention in many languages `context` comes first. In our case it is `project`.
Let's switch parameters.

Comment (acierto) [reply src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy:54]: Plus we can have such logic inside of WorkerUtil. 
For this place this code looks quite low level, same for license below

File: src/main/groovy/ai/digital/integration/server/tasks/worker/WorkerOverlaysTask.groovy [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import org.gradle.api.tasks.Copy
+
+import static ai.digital.integration.server.constant.PluginConstant.PLUGIN_GROUP
+
+class WorkerOverlaysTask extends DefaultTask {
+    static NAME = "workerOverlays"
+
+    static PREFIX = "worker"
+
+    static boolean shouldUnzip(File file) {
```
Comment (acierto) [inline src/main/groovy/ai/digital/integration/server/tasks/worker/WorkerOverlaysTask.groovy:16]: Let's create a util for that, it is copy pasted many times already

File: src/main/groovy/ai/digital/integration/server/tasks/worker/WorkerOverlaysTask.groovy [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            mustRunAfter DownloadAndExtractWorkerDistTask.NAME, SyncServerPluginsWithWorkerTask.NAME, SetWorkersLogbackLevelsTask.NAME
+
+            project.afterEvaluate {
+                WorkerUtil.getWorkers(project).each { Worker worker ->
+
+                    if (!worker.overlays.isEmpty() && !WorkerUtil.isExternalRuntimeWorker(worker, project)) {
+                        logger.warn("Overlays on the worker ${worker.name} are ignored because worker's runtime directory is same to the master.")
+                    } else {
+                        worker.overlays.each { Map.Entry<String, List<Object>> overlay ->
+                            def configurationName = "${PREFIX}${overlay.key.capitalize().replace("/", "")}"
```
Comment (acierto) [inline src/main/groovy/ai/digital/integration/server/tasks/worker/WorkerOverlaysTask.groovy:32]: The content of this for each is duplicated already 4 times, let's move it to some shared place.

File: src/main/groovy/ai/digital/integration/server/tasks/worker/SetWorkersLogbackLevelsTask.groovy (hunk 1 above)
Comment (vpugar-digital) [reply src/main/groovy/ai/digital/integration/server/tasks/worker/SetWorkersLogbackLevelsTask.groovy:12]: added LogbackUtil for common parts

File: src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy (hunk 4 above)
Comment (vpugar-digital) [reply src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy]: Changed log string

File: src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy (hunk 5 above)
Comment (vpugar-digital) [reply src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy]: Moved the project to the first place in arguments.

For the second comment, I changed that part and moved code to FileUtil, and now reuse that for copying files/dirs

File: src/main/groovy/ai/digital/integration/server/tasks/worker/WorkerOverlaysTask.groovy (hunk 6 above)
Comment (vpugar-digital) [reply src/main/groovy/ai/digital/integration/server/tasks/worker/WorkerOverlaysTask.groovy]: Moved everything that is repeating to the OverlaysUtil.

File: src/main/groovy/ai/digital/integration/server/tasks/worker/WorkerOverlaysTask.groovy (hunk 7 above)
Comment (vpugar-digital) [reply src/main/groovy/ai/digital/integration/server/tasks/worker/WorkerOverlaysTask.groovy]: Moved everything that is repeating to the OverlaysUtil.

File: src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy (hunk 3 above)
Comment (vpugar-digital) [reply src/main/groovy/ai/digital/integration/server/tasks/worker/SyncServerPluginsWithWorkerTask.groovy]: removed that line, and next line for chmod is executing only on bin folder

**Pull request merged**